### ISSUE TITLE
doesn't recognize some gem manpages

### ISSUE THREAD

**Issue by sunaku** [defunkt/gem-man#5]
Hello,

gem-man 0.2.0 does not recognize some gem manpages.  For instance:

```
> gem man -a
These gems have man pages:

kramdown 0.13.2
rdiscount 1.6.8
kramdown 0.10.0
```

In particular, my Inochi project isn't listed there, even though it has the same `man/` hierarchy as kramdown:

```
/home/stash/app/gems/gems/inochi-6.0.2/                                        
|+bin/                                                                         
|+lib/                                                                         
|~man/                                                                         
| `~man1/                                                                      
|   `-inochi.1                                                                                                        
`-LICENSE  
```

```
/home/stash/app/gems/gems/kramdown-0.13.2/                                     
|+benchmark/                                                                   
|+bin/                                                                         
|+data/                                                                        
|+doc/                                                                         
|+lib/                                                                         
|~man/                                                                         
| `~man1/                                                                      
|   `-kramdown.1    
```

file(1) has this to say about the .1 files:

```
> file `gem env gemdir`/gems/*/man/man1/*.1 | sed 's,^.*gems/,,'
detest-3.1.3/man/man1/detest.1:         troff or preprocessor input, UTF-8 Unicode text, with very long lines
ember-0.3.1/man/man1/ember.1:           troff or preprocessor input, ASCII text, with very long lines
inochi-6.0.2/man/man1/inochi.1:         Ruby module source, UTF-8 Unicode text, with very long lines
kramdown-0.10.0/man/man1/kramdown.1:    troff or preprocessor input, ASCII text
kramdown-0.13.2/man/man1/kramdown.1:    troff or preprocessor input, ASCII text
rumai-4.1.2/man/man1/rumai.1:           troff or preprocessor input, ASCII text, with very long lines
unicorn-3.6.0/man/man1/unicorn.1:       troff or preprocessor input, ASCII text
unicorn-3.6.0/man/man1/unicorn_rails.1: troff or preprocessor input, ASCII text
webgen-0.5.14/man/man1/webgen.1:        troff or preprocessor input, ASCII text
```

What's going on?  

Thanks for your consideration.

**Comment by defunkt** [comment]
`gem man inochi` works for me.

I also see it listed:

```
$ gem man -a
These gems have man pages:

mustache 0.99.3
shotgun 0.8
shotgun 0.9
inochi 6.0.2
foreman 0.10.1
bundler 1.0.11
ember 0.3.1
gem-man 0.2.0
```

Please reopen if this is still happening or you have more debug info. Thanks!

**Comment by sunaku** [comment]
The problem has become worse with the recent 1.8.2 rubygems update:

```
$ gem --version
1.8.2
$ gem man -a
These gems have man pages:

NOTE: Gem.source_index is deprecated, use Specification. It will be removed on or after 2011-11-01.
Gem.source_index called from /home/sunny/app/rubygems/gems/gem-man-0.2.0/lib/rubygems/commands/man_command.rb:65.
NOTE: Gem::SourceIndex#gems is deprecated with no replacement. It will be removed on or after 2011-11-01.
Gem::SourceIndex#gems called from /home/sunny/app/rubygems/gems/gem-man-0.2.0/lib/rubygems/commands/man_command.rb:65.
rdiscount 1.6.8
$ gem update gem-man
Updating installed gems
Nothing to update
$ gem list -l gem-man

*** LOCAL GEMS ***

gem-man (0.2.0)
```

**Comment by defunkt** [comment]
Ah. I'm still on an old version.

```
$ gem --version
1.5.2
```

If you're up to it I'd love a pull request, otherwise I'm gonna have to put this off until I upgrade rubygems myself. Thanks!

**Comment by sunaku** [comment]
This seems like a good starting point for adding RubyGems 1.8 compatibility:

``` irb
>> Gem.path.map {|path| Dir[path + '/gems/**/man/**.[1-9]'] }.flatten  
[
    [0] "/home/sunny/app/rubygems/gems/bundler-1.0.12/lib/bundler/man/gemfile.5",
    [1] "/home/sunny/app/rubygems/gems/bundler-1.0.13/lib/bundler/man/gemfile.5",
    [2] "/home/sunny/app/rubygems/gems/gem-man-0.2.0/man/gem-man.1",
    [3] "/home/sunny/app/rubygems/gems/rdiscount-1.6.8/man/markdown.7",
    [4] "/home/sunny/app/rubygems/gems/rdiscount-1.6.8/man/rdiscount.1"
]
```

**Comment by rymai** [comment]
Hi guys, I fixed this in #7.

I believe gem-man was broken since v1.6.0 of RubyGems actually!

**Comment by defunkt** [comment]
Thanks! Fixed by #7
